x/reports/keeper/msg_server.go

```diff
@@ -0,0 +1,285 @@
+		msg.SubspaceID,
+		reportID,
+		msg.ReasonID,
+		msg.Message,
+		msg.Target.GetCachedValue().(types.ReportTarget),
+		msg.Reporter,
+		ctx.BlockTime(),
+	)
+	err = k.ValidateReport(ctx, report)
+	if err != nil {
+		return nil, err
+	}
+
+	// Store the report
+	k.SaveReport(ctx, report)
+
+	// Update the id for the next report
+	k.SetNextReportID(ctx, msg.SubspaceID, report.ID+1)
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeyAction, sdk.MsgTypeURL(msg)),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.Reporter),
+		),
+		sdk.NewEvent(
+			types.EventTypeCreateReport,
+			sdk.NewAttribute(types.AttributeKeySubspaceID, fmt.Sprintf("%d", msg.SubspaceID)),
+			sdk.NewAttribute(types.AttributeKeyReportID, fmt.Sprintf("%d", report.ID)),
+			sdk.NewAttribute(types.AttributeKeyReasonID, fmt.Sprintf("%d", report.ReasonID)),
+			sdk.NewAttribute(types.AttributeKeyReporter, msg.Reporter),
+			sdk.NewAttribute(types.AttributeKeyCreationTime, report.CreationDate.Format(time.RFC3339)),
+		),
+	})
+
+	return &types.MsgCreateReportResponse{
+		ReportID:     report.ID,
+		CreationDate: report.CreationDate,
+	}, nil
+}
+
+// DeleteReport defines the rpc method for Msg/DeleteReport
+func (k msgServer) DeleteReport(goCtx context.Context, msg *types.MsgDeleteReport) (*types.MsgDeleteReportResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if the subspace exists
+	if !k.HasSubspace(ctx, msg.SubspaceID) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace with id %d not found", msg.SubspaceID)
+	}
+
+	// Check if the report exists
+	report, found := k.GetReport(ctx, msg.SubspaceID, msg.ReportID)
+	if !found {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "report with id %d not found inside subspace %d", msg.ReportID, msg.SubspaceID)
+	}
+
+	// Check the permission to delete reports
+	isModerator := k.HasPermission(ctx, msg.SubspaceID, msg.Signer, subspacestypes.PermissionManageReports)
+	canDelete := report.Reporter == msg.Signer && k.HasPermission(ctx, msg.SubspaceID, msg.Signer, subspacestypes.PermissionDeleteOwnReports)
+	if !isModerator && !canDelete {
+		return nil, sdkerrors.Wrap(subspacestypes.ErrPermissionDenied, "you cannot delete reports inside this subspace")
+	}
+
+	// Delete the post
+	k.Keeper.DeleteReport(ctx, msg.SubspaceID, msg.ReportID)
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeyAction, sdk.MsgTypeURL(msg)),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.Signer),
+		),
+		sdk.NewEvent(
+			types.EventTypeDeleteReport,
+			sdk.NewAttribute(types.AttributeKeySubspaceID, fmt.Sprintf("%d", msg.SubspaceID)),
+			sdk.NewAttribute(types.AttributeKeyReportID, fmt.Sprintf("%d", msg.ReportID)),
```

Comment:
Missing `ReasonID` here

Reply:
There is no reason to add the `reason_id` and `target` to the deleted report. All that's enough to know is simply the report it since it's being deleted and not created/edited. I don't see a usecase where knowing only the id of a deleted report is not enough